meta/saisanitycheck.c

```diff
@@ -1856,9 +1861,10 @@ void check_attr_acl_fields(
         case SAI_ATTR_VALUE_TYPE_ACL_ACTION_DATA_OBJECT_ID:
         case SAI_ATTR_VALUE_TYPE_ACL_ACTION_DATA_OBJECT_LIST:
 
-            if (md->objecttype != SAI_OBJECT_TYPE_ACL_ENTRY ||
-                    md->attrid < SAI_ACL_ENTRY_ATTR_ACTION_START ||
-                    md->attrid > SAI_ACL_ENTRY_ATTR_ACTION_END)
+            if (md->objecttype != SAI_OBJECT_TYPE_ACL_ENTRY  ||
```

Comment:
Those kind od changes needs to be introduced with changes that demonstrates need for relaxing this conditions

Reply:
@kcudnik I am trying to basically use the ACL changes from Jai's PR #1119 as an extension since that PR is not approved as yet.  I am copy pasting the relevant extension enum here.  

```
typedef enum _sai_acl_entry_attr_extensions_t
{
    SAI_ACL_ENTRY_ATTR_EXTENSIONS_RANGE_START = SAI_ACL_ENTRY_ATTR_END,

    /**
     * @brief ACL bind point for TAM object
     *
     * Bind (or unbind) a TAM object.
     *
     * @type sai_acl_action_data_t sai_object_id_t
     * @flags CREATE_AND_SET
     * @objects SAI_OBJECT_TYPE_TAM
     * @allownull true
     * @default disabled
     */
    SAI_ACL_ENTRY_ATTR_ACTION_TAM_OBJECT,

    SAI_ACL_ENTRY_ATTR_EXTENSIONS_RANGE_END

} sai_acl_entry_attr_extensions_t;
```

As you can see the sanity checker doesn't allow any extensions for ACL module since it explicitly checks for the value to be between ACTION_START and ACTION_END.  So, it prevents any extension since the extensions range starts after ACTION_END.

Let me know if this helps.

Reply:
my question here is why this change is separate from adding that extension acl entry ?

Reply:
Basically, this specific extension is a short term measure since Jai's PR is not merged in. The proposed change (to metachecker) on the other hand is intended to be generic.

Reply:
the problem im having (as in second added comment) is that not all acl entry extension attributes will be action/field, it could be regular acl_entry attribute like SAI_ACL_ENTRY_ATTR_ADMIN_STATE
lucky enough currently there are only 2 non field/action attributes and possibly extension attributes will be in that range